tests/filecheck/backend/riscv/riscv_register_allocation.mlir

```diff
@@ -42,8 +42,8 @@ riscv_func.func @main() {
 //   CHECK-LIVENESS-BLOCK-NAIVE-NEXT:        %{{\d+}} = riscv.mv %9 : (!riscv.reg<t0>) -> !riscv.reg<t0>
 //   CHECK-LIVENESS-BLOCK-NAIVE-NEXT:        riscv_scf.yield %{{\d+}} : !riscv.reg<t0>
 //   CHECK-LIVENESS-BLOCK-NAIVE-NEXT:      }
-//   CHECK-LIVENESS-BLOCK-NAIVE-NEXT:      %zero_0 = riscv.li 0 : () -> !riscv.reg<zero>
-//   CHECK-LIVENESS-BLOCK-NAIVE-NEXT:      %zero_1 = riscv.li 0 : () -> !riscv.reg<a0>
+//   CHECK-LIVENESS-BLOCK-NAIVE-NEXT:      %zero_1 = riscv.li 0 : () -> !riscv.reg<zero>
```

Comment:
but why starting from 1 now?

Reply:
Mmmh
We didn't change that logic; so it raises the question "how did it start at zero before" on my side :upside_down_face:

Reply:
Oooh no got it
The input has `%zero_0`, hence the name_hint used to be `zero_0` and printed back as such.
This is such a case where any other value name `zero` added in that IR would trigger the described bug.

Reply:
FWIW, I just tried offsetting the deduplication to start at 0 instead, it just makes even more filechecks fail, meaning more meaningless changes like this one in the same PR :upside_down_face:

Reply:
Tried starting from 0 instead, it just have more meaningless impacts like this on tests overall!